contracts/generator_controller/src/contract.rs

```diff
@@ -405,6 +423,29 @@ fn tune_pools(deps: DepsMut, env: Env) -> ExecuteResult {
         config.pools_limit,
     )?;
 
+    // Set allocation points for the main pool
+    if let Some(main_pool) = config.main_pool {
+        // Main pool may appear in the pool list thus we need to eliminate its contribution in the total VP
+        tune_info
+            .pool_alloc_points
+            .retain(|(pool, _)| pool != &main_pool.to_string());
+
+        let total_vp: Uint128 = tune_info
+            .pool_alloc_points
+            .iter()
+            .fold(Uint128::zero(), |acc, (_, vp)| acc + vp);
+        // Calculate main pool contribution.
+        // Example (30% for the main pool): VP + x = y, x = 0.3y => y = VP/0.7  => x = 0.3 * VP / 0.7,
```

Comment:
Should it be so or am I wrong?
For example: 
VP = 100
x = 10
y_vp = 110

x = 0.3 * y_vp = 33
y_new_vp = y_vp * 0.7 = 77

Reply:
`x` and `y` are unknowns. If VP=100, then x = 0.3 * 100/0.7 = 42,8571428571 and y = 142,8571428571.